stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/messages/view/MessageListScrollHelper.kt

```diff
@@ -0,0 +1,155 @@
+package io.getstream.chat.android.ui.messages.view
+
+import androidx.core.view.isVisible
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import com.getstream.sdk.chat.adapter.MessageListItem
+import io.getstream.chat.android.client.models.Message
+import io.getstream.chat.android.ui.messages.adapter.BaseMessageItemViewHolder
+import io.getstream.chat.android.ui.messages.adapter.MessageListItemAdapter
+import io.getstream.chat.android.ui.utils.extensions.cast
+import io.getstream.chat.android.ui.utils.extensions.isDeleted
+import io.getstream.chat.android.ui.utils.extensions.safeCast
+import kotlin.math.max
+
+internal class MessageListScrollHelper(
+    private val recyclerView: RecyclerView,
+    private val scrollButtonView: ScrollButtonView,
+    private val scrollToBottomButtonEnabled: Boolean,
+    private val callback: MessageReadListener,
+) {
+    internal var alwaysScrollToBottom = false
+
+    private val layoutManager: LinearLayoutManager by lazy { recyclerView.layoutManager as LinearLayoutManager }
+    private val adapter: MessageListItemAdapter by lazy { recyclerView.adapter as MessageListItemAdapter }
+
+    private var lastSeenMessageInChannel: MessageListItem? = null
+    private var lastSeenMessageInThread: MessageListItem? = null
+    private var isAtBottom = false
+    private var unreadCount = 0
+
+    private var currentList: List<MessageListItem> = emptyList()
+        get() {
+            return adapter.currentList
+        }
```

Comment:
I agree with the lint warning here - I think this was meant to be a `val`?

![image](https://user-images.githubusercontent.com/12054216/104746398-862c1180-574f-11eb-84b2-ee046defc3ee.png)